tests/tensorlake/test_graph_validation.py

```diff
@@ -110,8 +111,7 @@ def router2(a: int) -> Union[node1, node3]:
             )
 
             g.add_edge(node0, router)
```

Comment:
Should we make graph.add_edge private now? I thought users won't be calling it anymore. At the same time we need DocAI code to get updated to not use add_edge and use the new graph definition API (next=...).
Maybe we can add a TODO to make add_edge private (like _add_edge) once Doc AI migrated?